compiler/src/dmd/e2ir.d

```diff
@@ -4678,7 +4678,33 @@ elem *toElemCast(CastExp ce, elem *e, bool isLvalue, ref IRState irs)
             else if (rtl == RTLSYM.DYNAMIC_CAST &&
                      !cdto.isInterfaceDeclaration())
             {
+                //printf("cdfrom: %s cdto: %s\n", cdfrom.toChars(), cdto.toChars());
+                int level = 0;
+                auto b = cdto;
+                while (1)
+                {
+                    if (b == cdfrom)
+                        break;
+                    b = b.baseClass;
+                    if (!b)
+                    {
+//break;
+                        // did not find cdfrom, so cast fails, return null
+                        e = el_long(TYnptr, 0);
+                        return Lret(ce, e);
+                    }
+                    ++level;
+                }
+//printf("level = %d\n", level);
+		if (level == 0)
+		{
+		    return Lret(ce, e);	// cast to self
+		}
```

Comment:
you have some tabs here

Reply:
I'll fix it. Right now I'm trying to debug it. It isn't working.